store/commit_info.go

```diff
@@ -37,25 +37,44 @@ func (si StoreInfo) GetHash() []byte {
 
 // Hash returns the root hash of all committed stores represented by CommitInfo,
 // sorted by store name/key.
-func (ci CommitInfo) Hash() []byte {
+func (ci *CommitInfo) Hash() []byte {
 	if len(ci.StoreInfos) == 0 {
 		return nil
 	}
 
-	rootHash, _, _ := maps.ProofsFromMap(ci.toMap())
+	if len(ci.CommitHash) != 0 {
+		return ci.CommitHash
+	}
+
+	rootHash, _, _ := ci.GetStoreProof("")
 	return rootHash
 }
 
-func (ci CommitInfo) toMap() map[string][]byte {
-	m := make(map[string][]byte, len(ci.StoreInfos))
-	for _, storeInfo := range ci.StoreInfos {
-		m[storeInfo.Name] = storeInfo.GetHash()
+func (ci *CommitInfo) GetStoreProof(storeKey string) ([]byte, *CommitmentOp, error) {
```

Comment:
could we add a few test cases to make sure there are no regressions

Reply:
actually, it is not a part of this PR

Reply:
sounds good, that makes sense. thanks